cmd/login/cmd.go

```diff
@@ -296,11 +303,18 @@ func run(cmd *cobra.Command, argv []string) {
 		fedramp.Enable()
 	}
 
-	// Apply the default configuration details if not explicitly provided by the user:
-	gatewayURL, ok := ocm.URLAliases[env]
-	if !ok {
-		gatewayURL = env
+	gatewayURL, err := ocm.ResolveGatewayURL(env, cfg)
+	if err != nil {
+		r.Reporter.Errorf("Failed to resolve gateway URL: %v", err)
+		os.Exit(1)
 	}
+
+	// Make sure env is present in case of fedRAMP
+	if env == "" {
+		env = args.env
+	}
+
+	var ok bool
```

Comment:
Could you explain this to me? Fedramp has the same environments so i'm not sure of the reason we need this logic specifically for fedramp

Reply:
For example, line 247 in this command we use the env to get the right `LoginURL` for fedramp

Reply:
Hmm, right. Added check in the fedramp condition on line 247 itself.
We still need an empty check here because env would not be mandatory with this change and is defaulted to empty string